### PR TITLE
[C++20] Enable C++20 and Fixes

### PR DESCRIPTION
**Pull request by Akkadius**
We're currently on C++17 - this PR enables C++20

The compilation hit is roughly 10 seconds but nothing super crazy

**Features**

Here is a reference of features within C++20 if interested https://en.cppreference.com/w/cpp/20

Specifically, I'm interested in `designated initializers` as it makes code readability of loading data structures much cleaner.

**Before**

```
static struct my_member_t my_global_three[] = {
    { NULL, "one"},
    { NULL, "two"},
    { NULL, "three"},
};
```

**After**

```
static struct my_member_t my_global_three[] = {
    {.type = NULL, .name = "one"},
    {.type = NULL, .name = "two"},
    {.type = NULL, .name = "three"},
};
```

**Changes**

* `std::is_pod` is deprecated and swapped out for `std::is_standard_layout` in a class that is not currently in use in the code
* MSVC or Cmake does some strange stuff with preprocessors that are no longer available when enabling CPP20 that required some cmake adjustments.
* Silence many warnings in Windows that largely are noise and get ignored

**Linux**

Did some build testing for anyone else's notes

```
# full no-cache clean builds
# cpp17 with fmt 6.1.2 (3m1.387s)
# cpp20 with fmt 6.1.2 (3m25.174s)
# cpp17 with fmt 5.3.0 (3m7.506s)
# cpp20 with fmt 5.3.0 (3m35.234s)
# cpp20 with fmt 6.2.1 (3m8.326s)
# cpp20 with fmt 6.2.1 (3m30s)
# cpp20 with fmt 7.1.3 (3m3.906s)
# cpp20 with fmt 7.1.3 (3m7s)
# cpp17 with fmt 7.1.3 (2m56.933s)
# cpp20 with fmt 7.1.3 (3m6s)

# (cached smaller changed)
# cpp20 with fmt 7.1.3 (34s) (core header change)
# cpp20 with fmt 7.1.3 (26s) (core header change)
# cpp20 with fmt 7.1.3 (26s) (core header change)
# cpp17 with fmt 7.1.3 (34s) (core header change)
# cpp17 with fmt 7.1.3 (33s) (core header change)
# cpp17 with fmt 7.1.3 (34s) (core header change)
# cpp20 with fmt 7.1.3 (34s) (core header change)
# cpp20 with fmt 7.1.3 (35s) (core header change)
# cpp20 with fmt 7.1.3 (35s) (core header change)
```

**Windows**

```
========== Build: 22 succeeded, 0 failed, 2 up-to-date, 1 skipped ==========
```

### CODE REVIEW

File: common/strings.h [hunk 1]
```diff
@@ -44,9 +44,10 @@
 #include <tuple>
 #include <type_traits>
 
+#include <fmt/format.h>
+
 #ifndef _WIN32
```
Comment (Kinglykrab) [inline common/strings.h:49]: Do we need this anymore if there’s nothing inside it?

Comment (Akkadius) [reply common/strings.h:49]: It’s a ifndef with an else

Comment (Kinglykrab) [reply common/strings.h:49]: :+1:

**Comment by Akkadius** [comment]
Closing in favor of another branch

**Pull request closed without merging**